pydra/engine/tests/test_workflow.py

```diff
@@ -0,0 +1,376 @@
+def test_wf_ndst_5(plugin):
+    """ workflow with two tasks, outer splitter and combiner on tasks level"""
+    wf = Workflow(name="wf_ndst_5", input_spec=["x", "y"])
+    wf.add(multiply(name="mult", x=wf.lzin.x, y=wf.lzin.y).split(["x", "y"]))
+    wf.add(add2(name="add2", x=wf.mult.lzout.out).combine("mult.x"))
+    wf.inputs.x = [1, 2]
+    wf.inputs.y = [11, 12]
+    wf.set_output([("out", wf.add2.lzout.out)])
+    wf.plugin = plugin
+
+    with Submitter(plugin=plugin) as sub:
+        sub.run(wf)
+
+    # checking the results
+    while not wf.done:
+        sleep(1)
+    results = wf.result()
+
+    assert results.output.out[0] == [13, 24]
+    assert results.output.out[1] == [14, 26]
+
+
+@pytest.mark.parametrize("plugin", Plugins)
+def test_wf_st_6(plugin):
+    """ workflow with three tasks, third one connected to two previous tasks,
+        splitter on the workflow level
+    """
+    wf = Workflow(name="wf_st_6", input_spec=["x", "y"])
+    wf.add(add2(name="add2x", x=wf.lzin.x))
+    wf.add(add2(name="add2y", x=wf.lzin.y))
+    wf.add(multiply(name="mult", x=wf.add2x.lzout.out, y=wf.add2y.lzout.out))
+    wf.split(["x", "y"], x=[1, 2, 3], y=[11, 12])
+
+    wf.set_output([("out", wf.mult.lzout.out)])
+    wf.plugin = plugin
+
+    with Submitter(plugin=plugin) as sub:
+        sub.run(wf)
+
+    # checking the results
+    while not wf.done:
+        sleep(1)
+    results = wf.result()
+
+    assert len(results) == 6
+    assert results[0].output.out == 39
+    assert results[1].output.out == 42
+    assert results[5].output.out == 70
+
+
+@pytest.mark.parametrize("plugin", Plugins)
+def test_wf_ndst_6(plugin):
+    """ workflow with three tasks, third one connected to two previous tasks,
+        splitter on the tasks levels
+    """
+    wf = Workflow(name="wf_ndst_6", input_spec=["x", "y"])
+    wf.add(add2(name="add2x", x=wf.lzin.x).split("x"))
+    wf.add(add2(name="add2y", x=wf.lzin.y).split("x"))
+    wf.add(multiply(name="mult", x=wf.add2x.lzout.out, y=wf.add2y.lzout.out))
+    wf.inputs.x = [1, 2, 3]
+    wf.inputs.y = [11, 12]
+
+    wf.set_output([("out", wf.mult.lzout.out)])
+    wf.plugin = plugin
+
+    with Submitter(plugin=plugin) as sub:
+        sub.run(wf)
+
+    # checking the results
+    while not wf.done:
+        sleep(1)
+    results = wf.result()
+
+    assert len(results.output.out) == 6
+    assert results.output.out == [39, 42, 52, 56, 65, 70]
+
+
+#@pytest.mark.parametrize("plugin", Plugins)
+@pytest.mark.xfail(reason="WIP")
+def test_wfasnd_1(plugin="serial"):
+    """ workflow as a node
+        workflow-node with one task and no splitter
+    """
+    wfnd = Workflow(name="wfnd", input_spec=["x"])
+    wfnd.add(add2(name="add2", x=wfnd.lzin.x))
+    wfnd.set_output([("out", wfnd.add2.lzout.out)])
+    wfnd.inputs.x = 2
+
+    wf = Workflow(name="wf", input_spec=["x"])
+    wf.add(wfnd)
+    # TODO: is this the syntax
+    wf.set_output([("out", wf.wfnd.add2.lzout.out)])
```

Comment:
```suggestion
    wf.set_output([("out", wf.wfnd.lzout.out)])
```

Reply:
workflow nodes have inputs and outputs like anything else. the external workflow would not have access to any of the internal nodes outputs directly without calling `set_output` (i.e. line 360).

Reply:
yes, I was thinking about `wf.wfnd.lzout.out`, but `Workflow` object has no attribute `lzout`. So this should be changed

Reply:
since a workflow is a task, it should have all attributes of a task. hence it should work for a workflow object as well. lzout is done in getattr in the base class.

Reply:
it would only have it once `set_output` has been called.

Reply:
easy fix: workflow `getattr` should return this:

`return super().__getattr__(name)`

Reply:
but it doesn't work... I'm getting the error exactly when calling `set_output` (`wf.set_output([("out", wf.wfnd.lzout.out)]`). I will look at this later today

Reply:
after fixing workflow `getattr`?

Reply:
github site didn't refresh when I was writing, let me try the fix

Reply:
it works! Thank you! I will add more tests later today